core/src/services/github/core.rs

```diff
@@ -259,7 +261,34 @@ impl GithubCore {
                 let resp: ListResponse =
                     serde_json::from_reader(body.reader()).map_err(new_json_deserialize_error)?;
 
-                Ok(resp.entries)
+                Ok(resp)
+            }
+            StatusCode::NOT_FOUND => Ok(ListResponse::default()),
+            _ => Err(parse_error(resp).await?),
+        }
+    }
+
+    pub async fn list_with_recursive(&self, url: &str) -> Result<Vec<Tree>> {
```

Comment:
Please add comments for this API to make it clear that this API calls github's Tree based API that requires `git_url`.

Reply:
fixed